### PR TITLE
Switch to Visual Studio 2022

### PR DESCRIPTION
**Pull request by KVS85**
### Scope & Purpose

*(Please describe the changes in this PR for reviewers, motivation, rationale - **mandatory**)*

- [ ] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.9: *(Please link PR)*
  - [ ] Backport for 3.8: *(Please link PR)*
  - [ ] Backport for 3.7: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

**Comment by KVS85** [comment]
- [ ] https://jenkins.arangodb.biz/view/Windows/job/arangodb-ANY-windows-matrix/3618/

**Comment by KVS85** [comment]
- [ ] https://jenkins.arangodb.biz/view/Windows/job/arangodb-ANY-windows-matrix/3620/

**Review by goedderz — APPROVED** [review]

File: arangod/Replication2/ReplicatedState/LeaderStateManager.tpp [hunk 1]
```diff
@@ -53,7 +53,11 @@ void LeaderStateManager<S>::run() noexcept {
           data.updateInternalState(LeaderInternalState::kIngestingExistingLog);
           auto mux = Multiplexer::construct(self->logLeader);
           mux->digestAvailableEntries();
+#ifdef _MSC_VER                                   // circumventing bug in msvc
+          data.stream = mux->getStreamById<1>();  // TODO fix stream id
```
Comment (dothebart) [inline arangod/Replication2/ReplicatedState/LeaderStateManager.tpp:57]: please create a BTS for this todo.

**Review by dothebart — APPROVED** [review]
lgtm. Please create a BTS for the todo.

File: cmake/custom/ArangoDB_FindOpenSSL_WIN32.cmake [hunk 2]
```diff
@@ -56,8 +56,11 @@ if (1909 LESS ${MSVC_VERSION} AND ${MSVC_VERSION} LESS 1920)
 elseif (1919 LESS ${MSVC_VERSION} AND ${MSVC_VERSION} LESS 1930)
 # 1920-1929: VS 16.0 (v142 toolset)
     set(VS_VERSION "2019")
+elseif (1930 LESS ${MSVC_VERSION} AND ${MSVC_VERSION} LESS 1940)
```
Comment (mpoeter) [inline cmake/custom/ArangoDB_FindOpenSSL_WIN32.cmake:59]:
```suggestion
elseif (1929 LESS ${MSVC_VERSION} AND ${MSVC_VERSION} LESS 1940)
```

Comment (KVS85) [reply cmake/custom/ArangoDB_FindOpenSSL_WIN32.cmake:59]: Added separetely in https://github.com/arangodb/arangodb/commit/81ccd9e624d53654e4cfc4d09dee06ba451fe7bc.

File: arangod/Replication2/ReplicatedState/LeaderStateManager.tpp (hunk 1 above)
Comment (goedderz) [reply arangod/Replication2/ReplicatedState/LeaderStateManager.tpp:57]: why?

Comment (goedderz) [reply arangod/Replication2/ReplicatedState/LeaderStateManager.tpp:57]: That makes sense, I assumed @dothebart was referring to the TODO note `// TODO fix stream id`.

Comment (maierlars) [reply arangod/Replication2/ReplicatedState/LeaderStateManager.tpp:57]: please don't create at ticket. There is already a ticket in the CINFRA board that would remove this piece of code completely. https://arangodb.atlassian.net/browse/CINFRA-160

Comment (dothebart) [reply arangod/Replication2/ReplicatedState/LeaderStateManager.tpp:57]: ok, thanks @maierlars - I just wanted this not to be forgotten.

File: lib/Basics/debugging.cpp [hunk 3]
```diff
@@ -80,9 +80,13 @@ void TRI_TerminateDebugging(std::string_view message) {
...
     // the program but continues.
     auto f = []() noexcept {
-      // intentionally crashes the program!
+    // intentionally crashes the program!
+    // MSVC has this constexpr ctor deleted. But as Asan/Ubsan is anyway not an
+    // option for MSVC right now, just disable the code here.
+#ifndef _MSC_VER
       // cppcheck-suppress *
       return std::string(nullptr);
+#endif
```
[the remark this answers is not in the document]
Comment (mpoeter) [reply lib/Basics/debugging.cpp:89]: Yes, that should work.

**Pull request merged**